src/main/java/org/verapdf/wcag/algorithms/semanticalgorithms/ContrastRatioChecker.java

```diff
@@ -0,0 +1,177 @@
+        return 0.2126 * normalizedRed + 0.7152 * normalizeGreen + 0.0722 * normalizeBlue;
+    }
+
+    private double normalizeColorComponent(int colorComponent) {
+        double doubleColorComponent = colorComponent/255.;
+        return doubleColorComponent < 0.03928 ? doubleColorComponent/12.92 :
+                Math.pow(((doubleColorComponent + 0.055)/1.055), 2.4);
+    }
+
+    private double getContrastRatio(double first, double second) {
+        double l1 = Math.max(first, second);
+        double l2 = Math.min(first, second);
+        return (l1 + 0.05) / (l2 + 0.05);
+    }
+
+    private List<DataPoint> getLuminosityPresenceList(BufferedImage bim) {
+
+        int width = bim.getWidth();
+        int height = bim.getHeight();
+        Map<Color, DataPoint> colorMap = new HashMap<>();
+
+        for (int i = 0; i < width; i++) {
+            for (int j = 0; j < height; j++) {
+                int rgb = bim.getRGB(i, j); //always returns TYPE_INT_ARGB
+                int alpha = (rgb >> 24) & 0xFF;// not used for now, should remove?
+                int red =   (rgb >> 16) & 0xFF;
+                int green = (rgb >>  8) & 0xFF;
+                int blue =  (rgb      ) & 0xFF;
+                Color color = new Color(red, green, blue);
+                double relativeLuminosity = relativeLuminosity(color);
+
+                if (colorMap.containsKey(color)) {
+                    colorMap.get(color).totalOccurrence++;
+                } else {
+                    colorMap.put(color, new DataPoint(relativeLuminosity));
+                }
+            }
+        }
+
+        return new ArrayList<>(new TreeSet<>(colorMap.values()));
+    }
+
+    private List<DataPoint> findLocalMaximums(List<DataPoint> source)
+    {
+        int size = source.size();
+        List<DataPoint> localMaximums = new ArrayList<>();
+
+        if (source.get(0).totalOccurrence > source.get(1).totalOccurrence) {
+            localMaximums.add(source.get(0));
+        }
+
+        // Iterating over all points to check
+        // local maximum
+        for(int i = 1; i < size - 1; i++)
+        {
+            // Condition for local maximum
+            if ((source.get(i - 1).totalOccurrence < source.get(i).totalOccurrence) &&
+                    (source.get(i).totalOccurrence > source.get(i + 1).totalOccurrence)) {
+                localMaximums.add(source.get(i));
+            }
+        }
+
+        // Checking whether the last point is
+        // local maximum or none
+        if (source.get(size - 1).totalOccurrence > source.get(size- 2).totalOccurrence) {
+            localMaximums.add(source.get(size - 1));
+        }
+
+        return localMaximums;
+    }
+
+    private double[] get2MostPresentElements(List<DataPoint> source) {
+        double absoluteMaxPresent = -1;
+        double secondMaxPresent = -1;
+        int max = 0;
+        int secondMax = 0;
+
+        for (int i = 0; i < source.size(); i++) {
```

Comment:
replace with `for (DataPoint dataPoint : source) {`

Reply:
ok, done